src/blockchain.rs

```diff
@@ -0,0 +1,181 @@
+use crate::block::Block;
+use std::collections::HashMap;
+
+/// Indexes of chain attribute
+#[derive(Debug, Clone)]
+pub struct BlockchainIndex {
+    /// Vector of blocks
+    blocks: Vec<Block>,
+}
+
+impl BlockchainIndex {
+    /// Creates new `BlockahinIndex`
+    pub fn new() -> Self {
+        BlockchainIndex { blocks: vec![] }
+    }
+}
+
+/// The structure represents the state of the
+/// blockchain itself, including the blocks that are on the
+/// longest-chain as well as the material that is sitting off
+/// the longest-chain but capable of being switched over.
+#[derive(Debug, Clone)]
+pub struct Blockchain {
+    /// Index of `Block`s
+    index: BlockchainIndex,
+    /// Hash map of all `Block`s received, both on and off the longest chain
+    block_hash_longest_chain_hashmap: HashMap<[u8; 32], u8>,
+    /// Hash map of the ids of `Block` hashes
+    block_hash_block_id_hashmap: HashMap<[u8; 32], u64>,
```

Comment:
I think we should remove these data structures until they are needed.

If you think we need it very soon, can you give me an idea what they're needed for?